trappy/base.py

```diff
@@ -265,6 +265,24 @@ def generate_parsed_data(self):
 
             yield data_dict
 
+    def optimize_dataframe(self):
+        """Optimize memory footprint by setting minimal data types required by
+           each column"""
+        for col in self.data_frame.columns:
+            if 'int' in str(self.data_frame[col].dtype):
+                self.data_frame[col].apply(pd.to_numeric, downcast='unsigned')
+                continue
+            if 'float' in str(self.data_frame[col].dtype):
+                self.data_frame[col].apply(pd.to_numeric, downcast='float')
+                continue
+            # Convert string objects (pointer) to categories, only when we have
+            # a relatively limited number of unique values (50% of the rows)
+            num_unique_values = len(self.data_frame[col].unique())
+            num_total_values = len(self.data_frame[col])
+            if num_unique_values / num_total_values > 0.5:
```

Comment:
Mmm you can argue that you can save some memory even when you have >50% unique names, but that's not too relevant for us, I think.

Reply:
Right, that's a very coarse-grained heuristic... in the post linked by the commit message the guy suggest this value as something conservative. Still, I think it quite well fit most of our use cases where, with trace events, we have tons of repeated strings (e.g. task names).

Reply:
I'd be surprised if in sufficiently large dataframes you have this over the 50% mark to be honest